pkg/vulnerabilityreport/io.go

```diff
@@ -93,9 +96,68 @@ func (r *readWriter) FindByOwner(ctx context.Context, owner kube.Object) ([]v1al
 	if err != nil {
 		return nil, err
 	}
+	// no reports found for provided owner, look for reports in related replicaset
+	if len(list.Items) == 0 && (owner.Kind == kube.KindDeployment || owner.Kind == kube.KindPod) {
+		rsName, found := r.findRelatedReplicaset(ctx, owner)
+		if !found {
+			return nil, nil
+		}
+
+		list, err = r.starboardClientset.AquasecurityV1alpha1().VulnerabilityReports(owner.Namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: labels.Set{
+				kube.LabelResourceKind:      string(kube.KindReplicaSet),
+				kube.LabelResourceName:      rsName,
+				kube.LabelResourceNamespace: owner.Namespace,
+			}.String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return list.DeepCopy().Items, nil
 }
 
+// findRelatedReplicaset attempts to find the replicaset that is associated with
+// the given owner. If the owner is a Deployment, it will look for a replicaset
+// that is controlled by the Deployment. If the owner is a Pod, it will look for
+// the replicaset that owns the Pod.
+func (r *readWriter) findRelatedReplicaset(ctx context.Context, owner kube.Object) (rsName string, found bool) {
+	switch owner.Kind {
+	// find a replicaset controlled by the deployment
+	case kube.KindDeployment:
+		rsList, err := r.kubernetesClientset.AppsV1().ReplicaSets(owner.Namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return "", false
+		}
+		if len(rsList.Items) == 0 {
+			return "", false
+		}
+		for _, rs := range rsList.Items {
+			if len(rs.OwnerReferences) == 0 {
+				continue
+			}
+			for _, ownerRef := range rs.OwnerReferences {
+				if owner.Name == ownerRef.Name && *ownerRef.Controller {
+					return rs.Name, true
```

Comment:
I believe that we want to lookup the active ReplicaSet (aka current revision) here. For that I'd probably lookup and compare the `deployment.kubernetes.io/revision` annotation set on both Deployment and the ReplicaSet objects.

PS We're doing sth similar in Octant plugin [here](https://github.com/aquasecurity/starboard-octant-plugin/blob/main/pkg/plugin/model/repository.go#L116). However, the code cannot be shared easily as Octant has it's own Kubernetes client interfaces.

Reply:
Done. I would expect that checking `*ownerRef.Controller` and comparing revisions would give the same result, but I haven't verified this, so let's use revisions since that seems to work in starboard-octant-plugin.